pkg/cloud/cloudcheck/cloudcheck_processor.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cloudcheck
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+
+	"github.com/cockroachdb/cockroach/pkg/cloud"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
+	"github.com/cockroachdb/cockroach/pkg/sql/rowenc"
+	"github.com/cockroachdb/cockroach/pkg/sql/rowexec"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/pkg/errors"
+)
+
+type result struct {
+	ok        bool
+	error     string
+	readRate  int
+	writeRate int
+	canDelete bool
+}
+
+var flowTypes = []*types.T{types.Int, types.String, types.Bool, types.String, types.Int, types.Int, types.Bool}
+
+func checkStorage(ctx context.Context, store cloud.ExternalStorage) (result, error) {
+	const namePrefix = "cockroach-storage-test"
+	filename := fmt.Sprintf("%s-%d", namePrefix, rand.Int())
+	const chunkSize = 1 << 14    // 16kb
+	const payloadSize = 32 << 20 // 16mb
+
+	buf := make([]byte, chunkSize)
+	_, _ = rand.Read(buf)
```

Comment:
Maybe consider using a fixed seed tied to the job start time? That would guarantee byte-consistency across the nodes, which is maybe good?

Reply:
alternatively we could refill the buf with new random bytes every mb or something so the chunks aren't _all_ identical and really make it compression-immune. they shouldn't need to be byte-identical if they're distribution-identical and if they're not, we have bigger problems